src/live.rs

```diff
@@ -59,30 +63,52 @@ pub fn iso_ignition_embed(config: &IsoIgnitionEmbedConfig) -> Result<()> {
         }
     };
 
-    let cpio = make_cpio(&ignition)?;
-    if cpio.len() > embed.length {
-        bail!(
-            "Compressed Ignition config is too large: {} > {}",
-            cpio.len(),
-            embed.length
-        );
-    }
-    if !config.force {
-        // Ensure all zero bytes
-        embed.seek_to_start()?;
-        let mut buf = embed.new_buffer();
-        embed.read(&mut buf)?;
-        // compare to zeroed buffer
-        if buf != embed.new_buffer() {
-            bail!("This ISO image already has an embedded Ignition config; use -f to force.");
+    if config.output.as_deref() == USE_STDOUT {
```

Comment:
Let's break this into two separate `if` blocks, one to open files and one to do the writing, so we can avoid duplicating the `make_cpio()` call and the validity checks.  That'll require some fancy footwork in variable assignments, but should be doable.